app/models/grouping.rb

```diff
@@ -747,6 +741,143 @@ def get_random_incomplete(current_role)
 
   private
 
+  # Takes in a collection of results specified by +results+, and filters them using +filter_data+. Assumes
+  # +filter_data+ is not nil.
+  # +filter_data['annotationText']+ is a string specifying some annotation text to filter by. To avoid this filtering
```

Comment:
Don't repeat the "To avoid..." sentence for each option. Figure out a way of wording this documentation to avoid the repetition.